estuary-cdk/estuary_cdk/capture/common.py

```diff
@@ -648,6 +653,14 @@ async def _binding_incremental_task(
             elif isinstance(item, AssociatedDocument):
                 task.captured(item.binding, item.doc)
                 pending = True
+            elif item == Triggers.BACKFILL:
+                task.log.debug("incremental task triggered backfill")
+                task.checkpoint(
+                    ConnectorState(
+                        bindingStateV1={binding.stateKey: None}
```

Comment:
Very good point, thanks!

I added `task.stopping.event.set()` before the checkpoint, which stops all children of the connector task (that's what happens if there is an error). This way we don't need to error, but we will make sure the task exits once we checkpoint.

Reply:
Hmm I'm not sure that will totally prevent this, since the backfill/incremental tasks check for `stopping` at the ["top" of their loops](https://github.com/estuary/connectors/blob/502144546a2f069bfd5949310f4ffa843beb8f15/estuary-cdk/estuary_cdk/capture/common.py#L570-L604), and will still finish reading their current documents and outputting a checkpoint even if `stopping` is set while they are in the middle doing that. Indeed, as far as I understand it the point of the `stopping` event is specifically to allow other tasks to finish what they are doing a commit data prior to exiting when some other task has set the event in case of an error.

Reply:
An idea: Perhaps if one of these `BACKFILL` triggers is emitted from a task, it should set the `stopping` event and the emitted checkpoint should have a property that indicates the binding needs re-backfilled. Then when the connector starts back up the state for that binding can be cleared.